### PR TITLE
Optimize ExecutionStore#backendJobDescriptorKeyForNode. [BA-5767]

### PR DESCRIPTION
**Pull request by mcovarr**
before
```
size -> 100000: 72.38088547222222 ms
size -> 200000: 118.50121208333333 ms
size -> 300000: 175.1667145 ms
size -> 400000: 209.58718594444443 ms
size -> 500000: 280.38145025 ms
size -> 600000: 337.16847691666663 ms
size -> 700000: 367.9980148888889 ms
size -> 800000: 426.0341651944443 ms
size -> 900000: 539.8858220277777 ms
size -> 1000000: 548.2584588333334 ms
```

after
```
size -> 100000: 0.020786249999999992 ms
size -> 200000: 0.016441083333333335 ms
size -> 300000: 0.01688913888888889 ms
size -> 400000: 0.015165277777777776 ms
size -> 500000: 0.01751997222222222 ms
size -> 600000: 0.006870222222222223 ms
size -> 700000: 0.056051750000000004 ms
size -> 800000: 0.018104638888888885 ms
size -> 900000: 0.009031027777777775 ms
size -> 1000000: 0.002292388888888889 ms
```

### CODE REVIEW

File: engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/ExecutionStoreBenchmark.scala [hunk 1]
```diff
@@ -40,30 +40,36 @@ object ExecutionStoreBenchmark extends Bench[Double] with DefaultJsonProtocol {
...
+      val sizes: Gen[Int] = Gen.range("size")(from = 1000, upto = 10000, hop = 1000)
+      using(stores(sizes)) in { es =>
         es.update
       }
     }
+
+    measure method "backendJobDescriptorKeyForNode" in {
+      val sizes: Gen[Int] = Gen.range("size")(from = 100000, upto = 1000000, hop = 100000)
+      using(stores(sizes)) in { es =>
+        es.backendJobDescriptorKeyForNode(scatterCall)
```
Comment (cjllanwarne) [inline engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/ExecutionStoreBenchmark.scala]: Hmm, I wonder whether the test is not quite representative. If _all_ of the keys in the list are scatters and none are `BackendJobDescriptorKey ` then we presumably won't be exercising the equality check performance?

File: supportedBackends/google/pipelines/v2alpha1/src/main/scala/cromwell/backend/google/pipelines/v2alpha1/api/ActionBuilder.scala [hunk 2]
```diff
@@ -64,7 +64,9 @@ object ActionBuilder {
     val ContentTypeTextHeader = s"Content-Type: $contentTypeText"
   }
 
-  val cloudSdkImage = "google/cloud-sdk:slim"
+  // TODO revert this to google/cloud-sdk:slim once latest is unbroken
+  // TODO https://github.com/GoogleCloudPlatform/gsutil/issues/806
+  val cloudSdkImage = "google/google/cloud-sdk:251.0.0-slim"
```
Comment (cjllanwarne) [inline supportedBackends/google/pipelines/v2alpha1/src/main/scala/cromwell/backend/google/pipelines/v2alpha1/api/ActionBuilder.scala]: Is this a stowaway change? Seems unrelated to the rest of the PR?

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/stores/ExecutionStore.scala [hunk 3]
```diff
@@ -156,8 +156,8 @@ sealed abstract class ExecutionStore private[stores](statusStore: Map[JobKey, Ex
   lazy val store: Map[ExecutionStatus, List[JobKey]] = statusStore.groupBy(_._2).safeMapValues(_.keys.toList)
   lazy val queuedJobsAboveThreshold = queuedJobs > MaxJobsToStartPerTick
 
-  def keysForNode(node: GraphNode): Iterable[JobKey] = {
-    statusStore.keys collect { case k if k.node eq node => k }
+  def backendJobDescriptorKeyForNode(node: GraphNode): Option[BackendJobDescriptorKey] = {
+    statusStore.keys collectFirst { case k: BackendJobDescriptorKey if k.node eq node => k }
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/stores/ExecutionStore.scala]: improvements: 

- stop once the first match is found
- don't collect up a bunch of results that are not `BackendJobDescriptorKey`s which would only get thrown away later but in the meantime need to be hashed for insertion into an associative collection
- single pass

File: supportedBackends/google/pipelines/v2alpha1/src/main/scala/cromwell/backend/google/pipelines/v2alpha1/api/ActionBuilder.scala (hunk 2 above)
Comment (mcovarr) [inline supportedBackends/google/pipelines/v2alpha1/src/main/scala/cromwell/backend/google/pipelines/v2alpha1/api/ActionBuilder.scala]: unrelated to the `ExecutionStore` changes, this just happened to break at the time this work began

**Comment by cjllanwarne** [comment]
To save you from spelunking:
```
[0m[[0m[0minfo[0m] [0m[0m[31m- should convert a cloud sdk action *** FAILED *** (56 milliseconds)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  "docker run google/[google/cloud-sdk:251.0.0-]slim" was not equal to "docker run google/[cloud-sdk:]slim" (ActionBuilderSpec.scala:50)
```

File: engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/ExecutionStoreBenchmark.scala (hunk 1 above)
Comment (mcovarr) [reply engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/ExecutionStoreBenchmark.scala]: Discussed with faces, the existing test was a bit confusing. All the keys are actually `BackendJobDescriptorKey`s.
